lib/build.js

```diff
@@ -65,7 +65,8 @@ function getBlockedByIds(pipeline, job) {
     // Remove ~ prefix for job names
     blockedByNames = blockedByNames.map(name => name.replace('~', ''));
 
-    return pipeline.getJobs().then(pipelineJobs => {
+    // Without PR jobs
+    return pipeline.getJobs({ type: 'pipeline' }).then(pipelineJobs => {
```

Comment:
should we add this as a parameter instead and allow caller to decide whether to get or jobs or not?

Reply:
Also, will this work, for the time when PR job is blocked by another PR job —e.g., in case we restart a PR job when one is running

Reply:
This is just called from [start](https://github.com/screwdriver-cd/models/blob/4f62727856f659e8a529fd8f433b7ac9bc4c1835/lib/build.js#L460) and [stop](https://github.com/screwdriver-cd/models/blob/4f62727856f659e8a529fd8f433b7ac9bc4c1835/lib/build.js#L584), it seems to be enough to get the same result in both cases. So there seems to be no need to pass parameters to separate the processes.

The logic of the (PR) build blocking itself is the following part.
https://github.com/screwdriver-cd/models/blob/4f62727856f659e8a529fd8f433b7ac9bc4c1835/lib/build.js#L57

With this fix, pipelineJobs will be changed as follows. I found that in the case where `main` is in `blockedBy`, until now we also get the job ID of `PR-1:main`, so the commit build blocked by `main` was also blocked by the `PR-1:main` build. (FYI: [findIdsOfMatchedJobs](https://github.com/screwdriver-cd/models/blob/4f62727856f659e8a529fd8f433b7ac9bc4c1835/lib/build.js#L26-L45)) I think it is a strange phenomenon that commit builds are blocked by PR builds and vice versa. With this fix, PR builds are no longer completely bound to blockedBy.
```
pipelineJobs = [
  Job { ..., name: "main", ... },
//  Job { ..., name: "PR-1:main", ... }    //If set type:'pipeline', this line is omitted
]
```

Reply:
I think this issue is very much similar to https://github.com/screwdriver-cd/screwdriver/issues/2496